pkg/controllers/startup_policy.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+type StartupPolicyAction int64
+
+const (
+	Successful StartupPolicyAction = 1
+	Wait       StartupPolicyAction = 2
+	Resume     StartupPolicyAction = 3
+)
+
+func MatchStartupPolicy(js *jobset.JobSet, rjJobStatus jobset.ReplicatedJobStatus) StartupPolicyAction {
+
+	targetedReplicas := getReplicasFromName(js.Spec.ReplicatedJobs, rjJobStatus.Name)
+
+	if rjJobStatus.Suspended == targetedReplicas {
+		return Resume
```

Comment:
we shouldn't need this if we treat the unsuspend case like the creation one, both should be checking that the number of replicas are ready(or completed, success or failed as discussed below)

Reply:
updated.